agent/api/transitiondependency.go

```diff
@@ -17,5 +17,5 @@ type ContainerDependency struct {
 	SatisfiedStatus ContainerStatus `json:"SatisfiedStatus"`
 	// DependentStatus defines the status that cannot be reached until the
 	// resource satisfies the dependency
-	DependentStatus ContainerStatus `json:"DependentStatus"`
+	DependentStatus ContainerStatus `json:"DependentStatus,omitempty"`
```

Comment:
Is adding `omitempty` absolutely necessary here? Are there conditions in which it could break the JSON marshaling/unmarshaling logic? I'd rather not add this if it's only giving us marginal gains.

Reply:
We won't be using this field going forward, since `DependentStatus` will actually be the key in `TransitionDependenciesMap`. Hence added a `omitEmpty` since we are reusing `ContainerDependency` and this field would be empty. But we can just remove this field in here and take care of it in JSON Marshaler in a future PR?

Reply:
yeah, that'd be better.